### DISCUSSION TITLE
Can pkl be used like jsonlogic also ?

### DISCUSSION THREAD

**Discussion by arundotin** [apple/pkl#281] [General]
For some of our complex business rules, we use [jsonlogic](https://jsonlogic.com/), to get appropriate JSON Configuration. For instance : if we give `A` as input to our rules in jsonlogic, then it will validate some rules and give us a JSON output and it will give a different output for `B`

This is useful, but not readable, testable and things like that ..

we are looking for an alternative to jsonlogic. Wondering if pkl can be used for such scenarios

**Comment by bioball** [comment]
Pkl definitely does this, albeit, it works a little differently to JSON logic.

With Pkl, you would define your validation _and_ your data in Pkl, then have that produce JSON.

For example:

MyApplication.pkl
```
module MyApplication

port: UInt16(this > 1000)

output {
  renderer = new JsonRenderer {}
}
```

application.pkl
```
amends "MyApplication.pkl"

port = 1500
```

Then `pkl eval application.pkl` produces JSON to be consumed elsewhere.

You can also embed Pkl _directly_ into your application without going through JSON, if there is an appropriate language binding for it.

Right now, Pkl has limited support for validating external JSON.
